hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AUtils.java

```diff
@@ -1945,4 +1946,22 @@ public String toString() {
   public static String formatRange(long rangeStart, long rangeEnd) {
     return String.format("bytes=%d-%d", rangeStart, rangeEnd);
   }
+
+  private static final String BYTES_PREFIX = "bytes=";
+
+  public static Optional<Pair<Long, Long>> parseRange(String rangeHeader) {
+    // TODO: Surely, there is a better way.
+    if (rangeHeader != null && rangeHeader.startsWith(BYTES_PREFIX)) {
+      String[] values = rangeHeader.substring("bytes=".length()).split("-");
+      if (values.length == 2) {
+        try {
+          Long start = Long.parseUnsignedLong(values[0]);
+          Long end = Long.parseUnsignedLong(values[0]);
+          return Optional.of(Pair.of(start, end));
+        } catch(NumberFormatException e) {
+        }
+      }
+    }
+    return Optional.empty();
+  }
```

Comment:
Not sure if there's a better way.

Two edge cases:

- If we see more than one range, we currently return empty? Should we do that or should we handle more than one range?
- What happens if the range is unbounded?

Both of these aren't applicable to S3A today but just considering if we were to change that in the future. 🤔 I'm a little bit wary about promoting this immediately to S3AUtils, since we only have one use case. Looking at where its used - computing the size - things like unbounded ranges will make that really complex anyway. I'd be inclined to have this as a private method of AWSRequestAnalyzer for now, and if we do start needing ranges as components later then we can think about how to add it as a public method like this.

Reply:
HADOOP-18183 plans to add ranges as part of auditing so we will have a second use case that @sauraank will be adding soon.

HADOOP-18183 will be going into trunk on SDK V1 so doesn't need to do parsing yet, but we probably do need to think about the edge cases.

Reply:
Private method may be more appropriate. I originally added this to S3AUtils so it would be close to the `formatRange` method, which is used to format the header range we add to `GetObjectRequests`. 

Although, in the end, this is only used to report the size in logs. Maybe a simpler solution is to just report the range?

Reply:
Meh, I don't mind really. I do appreciate it being separate from the analyze method itself, be it private or in S3AUtils. Pick whichever option you think works best.